File: fix_lints.dart
import 'dart:io';

void main() async {
  final libDir = Directory('lib');
  if (!libDir.existsSync()) {
    print('lib directory not found');
    return;
  }

  // Find all .dart files recursively
  final files = libDir
      .listSync(recursive: true)
      .whereType<File>()
      .where((f) => f.path.endsWith('.dart'))
      .toList();

  for (final file in files) {
    String content = await file.readAsString();
    bool modified = false;

    // 1. avoid_print -> replace 'print(' with 'debugPrint('
    final printRegex = RegExp(r'(?<![a-zA-Z0-9_])print\(');
    if (printRegex.hasMatch(content)) {
      content = content.replaceAll(printRegex, 'debugPrint(');

      // Add foundation.dart if missing and requires debugPrint
      if (!content.contains("package:flutter/foundation.dart") &&
          !content.contains("package:flutter/material.dart") &&
          !content.contains("package:flutter/widgets.dart")) {
        content = "import 'package:flutter/foundation.dart';\n$content";
      }
      modified = true;
    }

    // 2. withOpacity(val) -> withValues(alpha: val)
    final opacityRegex = RegExp(r'\.withOpacity\(([^)]+)\)');
    if (opacityRegex.hasMatch(content)) {
      content = content.replaceAllMapped(opacityRegex, (match) {
        return '.withValues(alpha: ${match.group(1)})';
      });
      modified = true;
    }

    // 3. MaterialStateProperty -> WidgetStateProperty
    if (content.contains('MaterialStateProperty')) {
      content =
          content.replaceAll('MaterialStateProperty', 'WidgetStateProperty');
      modified = true;
    }

    // 4. MaterialState -> WidgetState
    if (content.contains('MaterialState.')) {
      content = content.replaceAll('MaterialState.', 'WidgetState.');
      modified = true;
    }

    // 5. Unintended HTML bracket error in app_config.dart
    if (file.path.contains('app_config.dart') &&
        content.contains('<YOUR_LARAVEL_URL_HERE>')) {
      content = content.replaceAll(
          '<YOUR_LARAVEL_URL_HERE>', 'YOUR_LARAVEL_URL_HERE');
      modified = true;
    }

    if (modified) {
      await file.writeAsString(content);
      print('Fixed: ${file.path}');
    }
  }

  print('Done applying custom fixes!');
}

File: fix_lints.js
const fs = require('fs');
const path = require('path');

function processDirectory(dirPath) {
    const entries = fs.readdirSync(dirPath, { withFileTypes: true });

    for (const entry of entries) {
        const fullPath = path.join(dirPath, entry.name);

        if (entry.isDirectory()) {
            processDirectory(fullPath);
        } else if (entry.isFile() && fullPath.endsWith('.dart')) {
            let content = fs.readFileSync(fullPath, 'utf8');

            let modified = false;

            // Replace print( with debugPrint(
            if (content.includes('print(')) {
                content = content.replace(/([^a-zA-Z0-9_])print\(/g, '$1debugPrint(');

                // Ensure standard foundation import exists if we added debugPrint
                if (!content.includes("import 'package:flutter/foundation.dart';")) {
                    content = "import 'package:flutter/foundation.dart';\n" + content;
                }
                modified = true;
            }

            // Replace .withOpacity(x) with .withValues(alpha: x)
            const opacityRegex = /\.withOpacity\(([^)]+)\)/g;
            if (opacityRegex.test(content)) {
                content = content.replace(opacityRegex, '.withValues(alpha: $1)');
                modified = true;
            }

            // Fix specific unintended html bracket issue in app_config.dart
            if (fullPath.includes('app_config.dart') && content.includes('<YOUR_LARAVEL_URL_HERE>')) {
                content = content.replace('<YOUR_LARAVEL_URL_HERE>', 'YOUR_LARAVEL_URL_HERE');
                modified = true;
            }

            if (modified) {
                fs.writeFileSync(fullPath, content, 'utf8');
                console.log(`Updated: ${fullPath}`);
            }
        }
    }
}

const libPath = path.join(__dirname, 'lib');
processDirectory(libPath);
console.log('Done.');
